packages/react-web/src/inbox/helpers.ts

```diff
@@ -0,0 +1,58 @@
+import { ProfileId, profileId as brandProfileId } from '@lens-protocol/react';
+import { invariant } from '@lens-protocol/shared-kernel';
+import { Conversation } from '@xmtp/react-sdk';
+
+const CONVERSATION_ID_PREFIX = 'lens.dev/dm';
+
+export function buildConversationId(profileIdA: string, profileIdB: string) {
+  const profileIdAParsed = parseInt(profileIdA, 16);
+  const profileIdBParsed = parseInt(profileIdB, 16);
+  return profileIdAParsed < profileIdBParsed
+    ? `${CONVERSATION_ID_PREFIX}/${profileIdA}-${profileIdB}`
+    : `${CONVERSATION_ID_PREFIX}/${profileIdB}-${profileIdA}`;
+}
+
+export function parseConversationId(conversationId: string): [string, string] {
+  const conversationIdWithoutPrefix = conversationId.replace(`${CONVERSATION_ID_PREFIX}/`, '');
+  const [profileIdA, profileIdB] = conversationIdWithoutPrefix.split('-');
+
+  invariant(profileIdA && profileIdB, 'Invalid conversation id');
+
+  return [profileIdA, profileIdB];
+}
+
+export function isLensConversation(
+  activeProfileId: string,
+  conversationId?: string,
+): conversationId is string {
+  if (conversationId && conversationId.includes(activeProfileId)) {
+    return true;
+  }
+  return false;
+}
+
+export function extractPeerProfileId(
+  conversationId: string | undefined,
+  activeProfileId: string,
+): ProfileId | undefined {
+  if (isLensConversation(activeProfileId, conversationId)) {
+    const [profileIdA, profileIdB] = parseConversationId(conversationId);
+    const result = profileIdA === activeProfileId ? profileIdB : profileIdA;
+
+    return brandProfileId(result);
+  }
+  return undefined;
+}
+
+export function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
+  return value !== null && value !== undefined;
+}
+
+/**
+ * Create a unique conversation ID based on sender/receiver addresses and
+ * context values
+ */
+export const createUniqueConversationId = (conversation?: Conversation): string =>
```

Comment:
wondering if we could avoid calling at all that function when `conversation` is not present?

Reply:
I copied it from here without thinking too much 🤦 it shouldn't be optional, thx for noticing
https://github.com/xmtp/xmtp-web/blob/main/packages/react-sdk/src/helpers/getConversationId.ts